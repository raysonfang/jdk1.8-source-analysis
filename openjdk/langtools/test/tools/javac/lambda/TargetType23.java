/*
 * @test /nodynamiccopyright/
 * @bug 8003280
 * @summary Add lambda tests
 *  check case of ambiguous method call with lambda whose body cannot
            complete normally
 * @compile/fail/ref=TargetType23.out -XDrawDiagnostics TargetType23.java
 */

class TargetType23 {

    interface Sam0 {
        void m();
    }

    interface Sam1 {
        int m();
    }

    interface Sam2 {
        String m();
    }

    interface Sam3<A> {
        A m();
    }


    void call(Sam0 s) { }
    void call(Sam1 s) { }
    void call(Sam2 s) { }
    <Z> void call(Sam3<Z> s) { }

    void test() {
        call(()-> { throw new RuntimeException(); }); //ambiguous - both call(Sam0), call(Sam2), call(Sam3) match
    }
}
