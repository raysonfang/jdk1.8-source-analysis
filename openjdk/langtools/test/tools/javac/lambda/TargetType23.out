TargetType23.java:35:9: compiler.err.ref.ambiguous: call, kindname.method, call(TargetType23.Sam2), TargetType23, kindname.method, <Z>call(TargetType23.Sam3<Z>), TargetType23
1 error
